### PR TITLE
T2s1 dump data

### PR DESCRIPTION
**Pull request by Yas-Shen**

### CODE REVIEW

File: functions.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
+<?php
+//set up db connection
+$servername = "localhost";
+$username = "root";
+$password = "password";
+$db = "vinyl-collection";
+
+function retrieveDb(): PDO
```
Comment (Steph-ski) [inline functions.php:7]: Add in a dock block for this function

File: functions.php [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+{
+    $db = new PDO('mysql:host=db; dbname=vinyl-collection', 'root', 'password');
+    $db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+    return $db;
+}
+
+$db = retrieveDb();
+//write sql stmnt and execute
```
Comment (Steph-ski) [inline functions.php]: remove comments

File: functions.php [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+$query = "SELECT `image`, `artist`, `album`, `year`, `genre`, `label-released` FROM `vinyl-collection`;";
+//prepare the stmnt
```
Comment (Steph-ski) [inline functions.php]: remove comments

File: functions.php [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+function retrieveDb(): PDO
+{
+    $db = new PDO('mysql:host=db; dbname=vinyl-collection', 'root', 'password');
+    $db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+    return $db;
+}
+
+$db = retrieveDb();
+//write sql stmnt and execute
+$query = "SELECT `image`, `artist`, `album`, `year`, `genre`, `label-released` FROM `vinyl-collection`;";
```
Comment (Steph-ski) [inline functions.php]: think this can be put into a function for displaying database

File: functions.php [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+//prepare the stmnt
+
+$stmnt = $db->prepare($query);
+
+$stmnt->execute();
+
+$result = $stmnt->fetchAll();
```
Comment (Steph-ski) [inline functions.php]: add in index.php file to have the procedural code in (to just leave functions in the functions.php file)

File: functions.php [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+$query = "SELECT `image`, `artist`, `album`, `year`, `genre`, `label-released` FROM `vinyl-collection`;";
+
+
+$stmnt = $db->prepare($query);
+
+$stmnt->execute();
+
+$result = $stmnt->fetchAll ();
+
+echo '<pre>';
```
Comment (HadertS) [inline functions.php]: Once you've functionalised the select statement as per Stephs comment, this should be move to index.php

**Review by HadertS — CHANGES_REQUESTED** [review]
Stephs comments have covered most things but i've added a small bit about moving out the echo to index.php

File: functions.php [hunk 7]
```diff
@@ -0,0 +1,34 @@
+<?php
+/**
+ * @param PDO $db the db conn
```
Comment (HadertS) [inline functions.php]: Orphaned docbloc, this should go with its function

File: functions.php [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+ * @return string
+ */
+
+$servername = "localhost";
```
Comment (HadertS) [inline functions.php]: functions should usually be declared before anything else, but I'm assuming these are WIP for the retrieveDb function as they're currently unused

File: functions.php (hunk 6 above)
Comment (Yas-Shen) [reply functions.php]: done

File: index.php [hunk 9]
```diff
@@ -0,0 +1,31 @@
+<?php
+
+require_once 'functions.php';
+
+$servername = "localhost";
```
Comment (nesclark) [inline index.php]: I think lines 5-8 can be removed as they are not doing anything

File: index.php [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+$username = "root";
+$password = "password";
+$db = "vinyl-collection";
+
+$db = retrieveDb();
+
```
Comment (nesclark) [inline index.php:8]: lines 11 - 19 are retrieving your data, these could be functionalised and put in your functions.php file

File: index.php [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+$query = "SELECT `image`, `artist`, `album`, `year`, `genre`, `label-released` FROM `vinyl-collection`";
+
+
+$stmnt = $db->prepare($query);
+
+$stmnt->execute();
+
+$result = $stmnt->fetchAll();
+
```
Comment (nesclark) [inline index.php]: lines 21 - 28 are displaying your data, you might find it easier to vardump this information

**Review by nesclark — CHANGES_REQUESTED** [review]
Almost there, just a few tweaks needed. Some code that can be removed and some parts that need functionalising. Vardumping would be a more straightforward way of displaying your data at the end.

File: index.php (hunk 11 above)
Comment (Yas-Shen) [reply index.php]: When I do that, it displays info in arrays.

Comment (Yas-Shen) [reply index.php]: Cracked it, just!

**Review by Joperes-ym — APPROVED** [review]

**Pull request merged**